zettarepl/replication/run.py

```diff
@@ -285,6 +327,15 @@ def run_replication_steps(step_templates: [ReplicationStepTemplate], observer=No
                 ignored_roots.add(step_template.src_dataset)
             continue
 
+        if is_immediate_target_dataset and step_template.dst_dataset not in step_template.dst_context.datasets:
+            # Target dataset does not exist, there is a chance that intermediate datasets also do not exist
+            for intermediate_dataset in get_intermediate_datasets(step_template.dst_dataset):
+                try:
+                    create_dataset(step_template.dst_context.shell, intermediate_dataset)
```

Comment:
Do we need to loop through every intermediate dataset? zfs create -p should create intermediate